src/NServiceBus.Core/Transports/Receiving.cs

```diff
@@ -46,36 +44,6 @@ protected internal override void Setup(FeatureConfigurationContext context)
             {
                 context.Settings.Get<QueueBindings>().BindReceiving(instanceSpecificQueue);
             }
-
-            var lazyReceiveConfigResult = new Lazy<TransportReceiveInfrastructure>(() => context.Settings.Get<TransportInfrastructure>().ConfigureReceiveInfrastructure());
```

Comment:
The reason I think was that we wanted to create the receive infrastructure only after the transport is fully initialized and not during the setup phase of the feature.